conftest.py

```diff
@@ -76,3 +77,8 @@ def pytest_collection_modifyitems(config, items):
                 reason=f"Test does not match the selected options: {', '.join(selected_markers)}"
             )
             item.add_marker(skip_this)
+
+
+@pytest.fixture(autouse=True)
+def set_random_seed():
+    np.random.seed(40)
```

Comment:
I think we ought to change this in PyBaMM as well. This can be:

```suggestion
@pytest.fixture(autouse=True)
def set_random_seed():
    return np.random.RandomState(40)
```

See: https://scientific-python.org/specs/spec-0007/

Reply:
This sets a [local random seed state](https://stackoverflow.com/a/22995942) (i.e. an object with that seed), but not the global state so I'm not sure how this could be easily propagated throughout the environment. Any thoughts?